### ISSUE TITLE
[Enhancement]: Not able to enable the RDS Data API using "enable_http_endpoint"

### ISSUE THREAD

**Issue by karthikanbu22** [hashicorp/terraform-provider-aws#37125]
### Description

We are trying to enable the RDS Data API using "enable_http_endpoint" but we are not able to do so. We are using Aurora PostgreSQL provisioned v2 clusters and Engine version 16.1. 

We raised AWS support case and we got below response:

###  the reason for Terraform not being able to enable data api could be due to the use of older unsupported ModifyCluster API call to set the http endpoint. The aws_rds_cluster resource might contain the older supported enable_http_endpoint attribute for aurora serverless v1 cluster, but seems it has not been updated to support serverless v2/provisioned clusters.###

If this is the case please enable this future for Aurora PostgreSQL serverless/provisioned v2 clusters

### Affected Resource(s) and/or Data Source(s)

Aurora PostgreSQL serverless/provisioned v2 

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @karthikanbu22 👋 Thank you for taking the time to raise this! Can you provide a bit more detail as to what errors (if any) you're receiving? I'd also like to check to see if the information in the proposed PR #37287 provides any clarity.

**Comment by arjentraas** [comment]
This command:
`aws rds modify-db-cluster --db-cluster-identifier my-cluster --enable-http-endpoint`

Outputs the resource description (as if you would use `aws rds describe-db-clusters --db-cluster-identifier my-cluster`)

**Comment by h-rasi** [comment]
Hi,
We are facing the same issue after migrating from Aurora Serverless v1 to v2. Setting the enable_http_endpoint attribute to true doesn't have any effect. The terraform plan command shows the changes, but terraform apply does not apply them. Every subsequent terraform plan continues to show the same changes.
We found that manually enabling the HTTP endpoint via the AWS console is a temporary workaround, as terraform apply no longer picks up the change after that.
This is critical for us due to the upcoming deadline to migrate all Aurora Serverless v1 databases to v2 by the end of the year.
Our configuration:
engine_mode: `provisioned`
engine: `aurora-postgresql`
engine_version: `13.12`

**Comment by yannickluzy-addactis** [comment]
Hi,

I am experiencing the same issue. Using enable_http_endpoint without snapshot_identifier works fine.

The documentation states:
enable_http_endpoint: This option does not work with any of these options specified: snapshot_identifier, replication_source_identifier, s3_import.
This is contradictory as AWS supports both enable_http_endpoint and snapshot_identifier configuration. Applying a snapshot on an endpoint database is a common practice. When run/rerunning Terraform, it shows the correct plan but fails to apply it without indication.

Currently, I use a null_resource to manually run aws rds enable-http-endpoint.

I understand this feature isn't supported. **It would be beneficial to enable both enable_http_endpoint and snapshot_identifier** to allow everyone to easily **upgrade to Aurora RDS V2**.

Our configuration:
aws_rds_cluster: using snapshot_identifier and enable_http_endpoint
PostgreSQL version: 16.1
aws_rds_cluster_instance: db.serverless

**Comment by wilkojd** [comment]
From the AWS docs.
Setting `EnableHttpEndpoint` in the body of a [ModifyDBCluster](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyDBCluster.html) only works for Aurora Serverless V1. For Serverless V2, the separate API command [EnableHttpEndpoint](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_EnableHttpEndpoint.html) must be used. This is not what the provider is doing.

Some redacted debug output:
```
aws_rds_cluster.cluster[0]: Modifying... [id=*****]
[INFO]  Starting apply for aws_rds_cluster.cluster[0]
[DEBUG] aws_rds_cluster.cluster[0]: applying the planned Update change
[DEBUG] provider.terraform-provider-aws_v5.61.0_x5: http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.9.3 (+https://www.terraform.io) terraform-provider-aws/5.61.0 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.55.5 (go1.22.5; darwin; arm64)" rpc.method=ModifyDBCluster
  http.request.body=
  | Action=ModifyDBCluster&ApplyImmediately=true&DBClusterIdentifier=*****&EnableHttpEndpoint=true&Version=2014-10-31
   
[DEBUG] provider.terraform-provider-aws_v5.61.0_x5: 
  http.response.body=
  | <ModifyDBClusterResponse xmlns="http://rds.amazonaws.com/doc/2014-10-31/"><ModifyDBClusterResult>.....<HttpEndpointEnabled>false</HttpEndpointEnabled>.....</ModifyDBClusterResult><ResponseMetadata><RequestId>*****</RequestId></ResponseMetadata></ModifyDBClusterResponse>
```

**Comment by YakDriver** [comment]
@yannickluzy-addactis @wilkojd There are two separate challenges. 1. `enable_http_endpoint` was not previously working for `provisioned` (and serverlessv2). This is fixed and will be available in v5.64.0. 2. The separate issue of `snapshot_identifier`, `replication_source_identifier`, and `s3_import` is based on observations (#37287) on does not come from the documentation. The AWS Provider does not prevent you attempting to use these with `enable_http_endpoint` so you are welcome to try. Please report back if you are able to use them together.
